diff -urN -x '*~' -x '*.orig' util-vserver-0.30.209-clean/scripts/functions util-vserver-0.30.209-poldek2/scripts/functions
--- util-vserver-0.30.209-clean/scripts/functions	2005-10-28 19:59:37.000000000 +0200
+++ util-vserver-0.30.209-poldek2/scripts/functions	2006-01-20 16:46:03.000000000 +0100
@@ -1017,15 +1055,17 @@
 
 function rpmFake.init
 {
-    local vdir ctx
+    local cdir vdir ctx
     
     vdir=$($_VSERVER_INFO "$1" VDIR)    || vdir="$1"
+    cdir=$($_VSERVER_INFO "$1" CFGDIR)  || cdir="$1"
     ctx=$($_VSERVER_INFO  "$1" CONTEXT) || ctx=
 
     test -d "$vdir" ||
 	panic $"Can not find chroot environment at '$vdir' for '$1'"
 
     _rpmFake.getCapFlags "$ctx"
+    generateOptions "$cdir"
 
     RPM_FAKE_CHROOT=$vdir
     RPM_FAKE_CTX=$ctx
@@ -1036,5 +1076,5 @@
     export RPM_FAKE_CHROOT RPM_FAKE_CTX RPM_FAKE_CAP RPM_FAKE_FLAGS
     
     LD_PRELOAD=$_RPM_FAKE_SO${LD_PRELOAD:+:$LD_PRELOAD} \
-	exec "$@"
+	exec $_CHBIND "${CHBIND_OPTS[@]}" "$@"
 }
diff -urN -x '*~' -x '*.orig' util-vserver-0.30.209-clean/scripts/vrpm-preload util-vserver-0.30.209-poldek2/scripts/vrpm-preload
--- util-vserver-0.30.209-clean/scripts/vrpm-preload	2005-03-21 21:03:30.000000000 +0100
+++ util-vserver-0.30.209-poldek2/scripts/vrpm-preload	2006-01-20 16:41:15.000000000 +0100
@@ -24,6 +24,7 @@
     exit 1
 }
 . "$UTIL_VSERVER_VARS"
+. $__PKGLIBDIR/vserver.functions
 . "$_LIB_FUNCTIONS"
 
 for bin in `which rpm` /bin/rpm /usr/lib/rpm/rpm${RPM_FLAVOR:-i} ""; do
