diff -urNp -x '*.orig' util-vserver-0.30.216-pre3126.org/kernel/network.h util-vserver-0.30.216-pre3126/kernel/network.h
--- util-vserver-0.30.216-pre3126.org/kernel/network.h	2016-02-24 19:09:17.000000000 +0100
+++ util-vserver-0.30.216-pre3126/kernel/network.h	2021-03-12 21:50:27.954385798 +0100
@@ -6,7 +6,7 @@
 
 #define MAX_N_CONTEXT	65535	/* Arbitrary limit */
 
-#define NB_IPV4ROOT	16
+#define NB_IPV4ROOT	128
 
 
 /* network flags */
diff -urNp -x '*.orig' util-vserver-0.30.216-pre3126.org/src/chbind.c util-vserver-0.30.216-pre3126/src/chbind.c
--- util-vserver-0.30.216-pre3126.org/src/chbind.c	2016-02-24 19:09:17.000000000 +0100
+++ util-vserver-0.30.216-pre3126/src/chbind.c	2021-03-12 21:50:27.954385798 +0100
@@ -262,7 +262,7 @@ int main (int argc, char *argv[])
       case CMD_NID		:  WRITE_MSG(2, "WARNING: --nid is not supported by this version\n"); break;
       case CMD_IP		:
 	if (nbaddrs>=nb_ipv4root) {
-	  WRITE_MSG(2, "Too many IP numbers, max 16\n");
+	  WRITE_MSG(2, "Too many IP numbers, max 128\n");
 	  exit(wrapper_exit_code);
 	}
 	readIP(optarg, ips+nbaddrs, &bcast);
