diff -Nur util-vserver-0.30.215/distrib.org/defaults/rpm/macros util-vserver-0.30.215/distrib/defaults/rpm/macros
--- util-vserver-0.30.215/distrib.org/defaults/rpm/macros	2008-03-17 08:59:48.000000000 +0000
+++ util-vserver-0.30.215/distrib/defaults/rpm/macros	2009-02-22 13:48:58.314083575 +0000
@@ -1,4 +1,5 @@
 # Because of https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=52725
 # you have to use at least one language beside 'C'
-%_install_langs		C:de:en:es:fr
+%_install_langs         C:en:pl
 %_netsharedpath		/dev:/etc/rc.d/init.d/halt:/sbin/new-kernel-pkg:/usr/bin/rhgb-client
+%_check_dirname_deps    0
diff -Nur util-vserver-0.30.215/distrib.org/redhat/initpre util-vserver-0.30.215/distrib/redhat/initpre
--- util-vserver-0.30.215/distrib.org/redhat/initpre	2008-03-17 08:59:48.000000000 +0000
+++ util-vserver-0.30.215/distrib/redhat/initpre	2009-02-22 13:46:50.000936159 +0000
@@ -19,7 +19,7 @@
 . "$2"
 
 ## Fix/workaround the misclassification of %_netsharedpath's parent directories
-mkdir -p "$vdir"/{etc/rc.d/init.d,/sbin,/usr/bin}
+mkdir -p "$vdir"/{etc/rc.d/init.d,/sbin,/usr/bin,root/tmp}
 
 ## Workaround broken kernel %post scripts
 ln -s ../bin/true "$vdir"/sbin/new-kernel-pkg
