--- util-vserver-0.30.216-r2837.orig/distrib/Makefile.am	2009-06-22 17:36:52.000000000 +0200
+++ util-vserver-0.30.216-r2837/distrib/Makefile.am	2009-06-22 17:49:36.000000000 +0200
@@ -85,7 +86,8 @@
 			   gentoo/init-vserver.sh gentoo/net.vserver \
 			   gentoo/reboot.sh gentoo/shutdown.sh\
 			   pld-ac/pkgs/01 pld-ac/rpm/macros \
-			   pld-th/pkgs/01 pld-th/rpm/macros
+			   pld-th/pkgs/01 pld-th/rpm/macros \
+			   tld/pkgs/01 tld/rpm/macros
 
 nobase_confdistrib_DATA	=  rh9/apt/sources.list \
 			   fc1/apt/sources.list \
@@ -95,7 +99,8 @@
 			   f7/apt/sources.list \
 			   suse91/apt/sources.list \
 			   pld-ac/poldek/repos.d/pld.conf \
-			   pld-th/poldek/repos.d/pld.conf
+			   pld-th/poldek/repos.d/pld.conf \
+			   tld/poldek/repos.d/tld.conf
 
 AM_INSTALLCHECK_STD_OPTIONS_EXEMPT \
 			=  $(nobase_distrib_SCRIPTS)
--- util-vserver-0.30.216-r2837.orig/distrib/tld/pkgs/01	1970-01-01 01:00:00.000000000 +0100
+++ util-vserver-0.30.216-r2837/distrib/tld/pkgs/01	2009-06-22 17:48:26.000000000 +0200
@@ -0,0 +1,6 @@
+basesystem
+filesystem
+glibc-misc
+issue
+rpm-base
+vserver-packages
--- util-vserver-0.30.216-r2837.orig/distrib/tld/poldek/repos.d/tld.conf	1970-01-01 01:00:00.000000000 +0100
+++ util-vserver-0.30.216-r2837/distrib/tld/poldek/repos.d/tld.conf	2009-06-22 17:48:26.000000000 +0200
@@ -0,0 +1,83 @@
+# Titanium Linux Distribution
+# http://www.tld-linux.org/
+
+_arch   = %ARCH%
+_prefix = ftp://ftp.tld-linux.org/TLD
+
+############################################################
+
+###
+# Titanium (Ti) Stable
+[source]
+type   = pndir
+name   = ti
+path   = %{_prefix}/stable/main/%{_arch}/RPMS/
+
+[source]
+type   = pndir
+name   = ti
+path   = %{_prefix}/stable/main/noarch/RPMS/
+
+###
+# Titanium (Ti) Stable, test packages
+[source]
+type   = pndir
+name   = ti-test
+path   = %{_prefix}/stable/test/%{_arch}/RPMS/
+auto   = no
+autoup = no
+
+[source]
+type   = pndir
+name   = ti-test
+path   = %{_prefix}/stable/test/noarch/RPMS/
+auto   = no
+autoup = no
+
+###
+# Titanium (Ti) Stable, archived packages
+[source]
+type   = pndir
+name   = ti-archive
+path   = %{_prefix}/stable/archive/%{_arch}/RPMS/
+auto   = no
+
+[source]
+type   = pndir
+name   = ti-archive
+path   = %{_prefix}/stable/archive/noarch/RPMS/
+auto   = no
+
+############################################################
+
+###
+# Titanium (Ti) Devel
+[source]
+type   = pndir
+name   = ti-dev
+path   = %{_prefix}/devel/main/%{_arch}/RPMS/
+auto   = no
+autoup = no
+
+[source]
+type   = pndir
+name   = ti-dev
+path   = %{_prefix}/devel/main/noarch/RPMS/
+auto   = no
+autoup = no
+
+###
+# Titanium (Ti) Devel, test packages
+[source]
+type   = pndir
+name   = ti-dev-test
+path   = %{_prefix}/devel/test/%{_arch}/RPMS/
+auto   = no
+autoup = no
+
+[source]
+type   = pndir
+name   = ti-dev-test
+path   = %{_prefix}/devel/test/noarch/RPMS/
+auto   = no
+autoup = no
--- util-vserver-0.30.216-r2837.orig/distrib/tld/rpm/macros	1970-01-01 01:00:00.000000000 +0100
+++ util-vserver-0.30.216-r2837/distrib/tld/rpm/macros	2009-06-22 17:48:26.000000000 +0200
@@ -0,0 +1,6 @@
+# Because of https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=52725
+# you have to use at least one language beside 'C'
+%_install_langs		C:en:pl
+%_netsharedpath		/dev:/proc
+%_tmppath			/tmp
+%_host_os			%_os
--- util-vserver-0.30.216-r2837.orig/scripts/functions	2009-06-22 17:36:35.000000000 +0200
+++ util-vserver-0.30.216-r2837/scripts/functions	2009-06-22 17:48:26.000000000 +0200
@@ -770,6 +785,8 @@
 	style=mandrake
     elif test -e "$_pgs_vdir"/etc/pld-release; then
 	style=pld
+    elif test -e "$_pgs_vdir"/etc/tld-release; then
+	style=pld
     elif test -e "$_pgs_vdir"/etc/debian_version; then
 	style=debian
     elif test -e "$_pgs_vdir"/etc/SuSE-release; then
--- util-vserver-0.30.216-r2837.orig/scripts/vserver-build.functions	2009-06-22 17:36:35.000000000 +0200
+++ util-vserver-0.30.216-r2837/scripts/vserver-build.functions	2009-06-22 17:48:26.000000000 +0200
@@ -170,6 +170,8 @@
 	    set -- $(cat /etc/pld-release)
 	    DISTRIBUTION=pld-$(echo $2 | tr A-Z a-z)
 	    IFS=$old_IFS
+	elif test -e /etc/tld-release; then
+	    DISTRIBUTION=tld
 	elif test -n "$1"; then
 	    DISTRIBUTION=$1
 	else
