diff -ur util-vserver-0.30.207/m4/ensc_dietlibc.m4 util-vserver-0.30.207.diet/m4/ensc_dietlibc.m4
--- util-vserver-0.30.207/m4/ensc_dietlibc.m4	2005-04-12 21:23:28.000000000 +0200
+++ util-vserver-0.30.207.diet/m4/ensc_dietlibc.m4	2005-06-30 19:41:15.000000000 +0200
@@ -78,7 +78,7 @@
 		       esac],
 		      [which "$DIET" >/dev/null 2>/dev/null && use_dietlibc=detected || use_dietlibc=detected_no])
 
-	if test "$use_dietlibc" = detected -a "$2"; then
+	if test "$use_dietlibc" != detected_no -a "$use_dietlibc" != forced_no -a "$2"; then
 	    _dietlibc_ver=$($DIET -v 2>&1 | sed '1p;d')
 	    _dietlibc_ver=${_dietlibc_ver##*diet version }
 	    _dietlibc_ver=${_dietlibc_ver##*dietlibc-}
