--- util-vserver-0.30.207/scripts/vsysvwrapper.orig	2005-03-21 21:03:31.000000000 +0100
+++ util-vserver-0.30.207/scripts/vsysvwrapper	2005-05-07 02:17:32.000000000 +0200
@@ -18,11 +18,12 @@
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 
 # Helper script for all the v_xxx scripts
-: ${UTIL_VSERVER_VARS:=/usr/lib/util-vserver/util-vserver-vars}
-test -e "$UTIL_VSERVER_VARS" || {
-    echo $"Can not find util-vserver installation (the file '$UTIL_VSERVER_VARS' would be expected); aborting..." >&2
-    exit 1
-}
+
+[ -n "$UTIL_VSERVER_VARS" ] || UTIL_VSERVER_VARS=/usr/lib/util-vserver/util-vserver-vars
+if [ ! -e "$UTIL_VSERVER_VARS" ] ; then
+	echo "Can not find util-vserver installation (the file '$UTIL_VSERVER_VARS' would be expected); aborting..." >&2
+	exit 1
+fi
 . "$UTIL_VSERVER_VARS"
 
 if [ $# = 0 ] ; then
@@ -40,6 +41,5 @@
 	do
 		IPOPT="$IPOPT --ip $oneip"
 	done
-	echo exec $CHBIND $IPOPT /etc/init.d/$SERVICE "$@"
-	exec $CHBIND $IPOPT /etc/init.d/$SERVICE "$@"
+	exec $CHBIND $IPOPT /etc/rc.d/init.d/$SERVICE "$@"
 fi
