diff -urN util-vserver-0.30.216-r2834.orig/distrib/f10/pkgs/01 util-vserver-0.30.216-r2834/distrib/f10/pkgs/01
--- util-vserver-0.30.216-r2834.orig/distrib/f10/pkgs/01	1970-01-01 01:00:00.000000000 +0100
+++ util-vserver-0.30.216-r2834/distrib/f10/pkgs/01	2009-04-27 17:02:41.000000000 +0200
@@ -0,0 +1 @@
+glibc
diff -urN util-vserver-0.30.216-r2834.orig/distrib/f10/pkgs/02 util-vserver-0.30.216-r2834/distrib/f10/pkgs/02
--- util-vserver-0.30.216-r2834.orig/distrib/f10/pkgs/02	1970-01-01 01:00:00.000000000 +0100
+++ util-vserver-0.30.216-r2834/distrib/f10/pkgs/02	2009-04-27 17:02:41.000000000 +0200
@@ -0,0 +1,3 @@
+--reinstall
+
+filesystem
diff -urN util-vserver-0.30.216-r2834.orig/distrib/f10/pkgs/03 util-vserver-0.30.216-r2834/distrib/f10/pkgs/03
--- util-vserver-0.30.216-r2834.orig/distrib/f10/pkgs/03	1970-01-01 01:00:00.000000000 +0100
+++ util-vserver-0.30.216-r2834/distrib/f10/pkgs/03	2009-04-27 17:02:41.000000000 +0200
@@ -0,0 +1,2 @@
+coreutils
+initscripts
diff -urN util-vserver-0.30.216-r2834.orig/distrib/f10/pubkeys/RPM-GPG-KEY-fedora-10-primary util-vserver-0.30.216-r2834/distrib/f10/pubkeys/RPM-GPG-KEY-fedora-10-primary
--- util-vserver-0.30.216-r2834.orig/distrib/f10/pubkeys/RPM-GPG-KEY-fedora-10-primary	1970-01-01 01:00:00.000000000 +0100
+++ util-vserver-0.30.216-r2834/distrib/f10/pubkeys/RPM-GPG-KEY-fedora-10-primary	2009-01-22 00:52:16.000000000 +0100
@@ -0,0 +1,40 @@
+-----BEGIN PGP PUBLIC KEY BLOCK-----
+Version: GnuPG v1.4.9 (GNU/Linux)
+
+mQGiBEi12/MRBADL6fpbRejYViiVzY/Y0T6pmgChOykrsq1oc9mX9+ss1lZZit1z
+DRrgvOYG67wnRQaskXnKulw3XoNEAT/OjCNzDwrS3gMbHaz+T9wQN5WeO52ihozp
+ZLo/aNnIW9a7W5KxMyrmDobo5919wjiHXFXbVyH5kgm1B36ZG09/HlwtpwCgyW/Z
+qczIsg9OhIrwYZRUJ8BHNdED/R6SFCF2p4AqMkpo6bO9+N5BLvTRPythuWyfTpOa
+XGLPDzkDisIW72DDmCrVI4RQnLW5u0xko7L1o8zwOmmFg6C8HLeeHpfg40MVLDDJ
+OtX1UTPmKms6eqKYWn15266y8Bga5QnVNGMRRPXK0YJhog8K4h7MV5i/B7F2Byhu
+rcT2A/42Ikj3qoA7mxIEgU3DDrxboKWZghAIWaMn8xqIl0/fIMmx/x3k6lDakQRq
++E6kdTvrWxsKbu5ESTzCDIjhHyVIECLGJ+EbB+etIa3N1+ek3gKEBcDHBtvbdc57
+NcYCWvDvL3O/SNpkKetm4kzrvAqL0Q6mvmGjSY4odoIPczhSlbQmRmVkb3JhICgx
+MCkgPGZlZG9yYUBmZWRvcmFwcm9qZWN0Lm9yZz6IYAQTEQIAIAUCSLXb8wIbAwYL
+CQgHAwIEFQIIAwQWAgMBAh4BAheAAAoJEL8ib8xOv8JzaaEAnjrJTa3QCZr9C91r
+MYJ2uNQvMQ5rAKCfqGK2Q2AVxK3pK1QHtc972g5gTrkEDQRItdvzEBAAz4E9yA4G
+epLon5F70Ak4/OgpJqHvXFN2xMGazpExkh+3/bjil3/J6rucXtLutPP4kIIu6LgW
+sbDF3HUafWpeOQI7YxlMwLXTfZ0Mevrsm4wEBSYoC8U4zygVyz2m6Rz2i62hbj/W
+AYIayW2sxg7VwbFoJxjjZfw2d6jTNQW9EI3GVffYUN+V7BSeQw7Y6v18pxFUxCur
+qOGXwNSUdUUmdh1h2O+CmvdSiYJApEU/fBHNioTDNVynTVN1ayobTs/5xANnpG+K
+7Lhiz2HrY6tTv34DaDMwv2aMoDxq5BTfi+H7rTzXlpF5fAjHNGsXQIJjKhEgpkHo
+hr+in0s6CsDpERpleS8FKIhjNwcGksoviVpEItnsHTOV0yOV5ZiuZpZ8M6ZS4Vf/
+JhvRCK2CeCkeyaAgxsGngN/QeTHE4e35m/dqUnBrUKRNkpcJ8jJO11eVNj0Xe/EG
+Vcy345TvIUCYG6kr1wJJt0xJm1wDricUHnR2X5motCOT/k6BF0ucofvcy1Ou62+N
+DKgI0E/o3BMo9XfPJVteheMOhwRMfNwmbFdr13rKk/GZyVeZmLwlFx1byH5r4J5T
+4vE9JmgeM2bVu/TiMFlBszQM39xcE3SycjX5knBEglzad23kzyIerlv8RZUdQJ/L
+qfmRXc+0x0H565YLwGC79XCKUp7dlGBYhL8ABAsP/1DrF/SzkNki1KWMeK8DP34q
+GBDWtUIVTiMqHZclUylHRT3bp2Ae2/+xy7NAy/Y/G5aZqkAMvaqiG3ps5LDVJo+1
+WAQBOgnwrZtVn6FU6H59a4h79hKoFoBEI1gd9XKzzgA+RF4cEHJiuFQh2boxdr7/
+n64tzrS6HkUcD15wIcYoyl6iOHBfbazbyZifeKMu0USkrfzLRLPec2V3p9bXqg1W
+h+SBCpD9uhSzBeKs636ipOL3WZ9UUMQMPNKyyFxx9cKcZY9aP0Leub2kvnFIZiG7
+ljbRP1dNJlhimTxTjZzfOicxu5MaSTyIlg9ZVEPXOmPwi/a8POP82pLZYZDi8Scx
+zQFNi+22VdSrQkj6Jkan3GoqBOBoSv9iQ7EJp7wpMBRhoacW3wtJ4m0+cAR8IX9X
+6DTklAU9zoZzdvg063GnjgPCi3Ohwl2jvM1NUcn4vdI5UUTd7DAPaysueJPqfFCg
+CSgrOcCyfQL4vL0iRp82NNQtUS9ZdyUw8t6w/GuFnsymBSTqt2t1n+mhS7RAejaV
+1QV/KmwijbaVh2LjTP0YzPWZcBy8XHzOsnuWufFmRTgFOUmWT9o82RSde6ubvnDN
+WfWWGzriIpJd3JCBpkVb2aiEkU4958Ai36I/splnxOrAJZV/4ZIHHnGMwn23/DNh
+gX4TCgPMjjjT0hmTfDpliEkEGBECAAkFAki12/MCGwwACgkQvyJvzE6/wnM+4wCe
+KkKsIfsmCT0MpuYxHh4vRS/ix58AoLfjES8VpN+3ku9hC3QT97Fa/bK4
+=s6iX
+-----END PGP PUBLIC KEY BLOCK-----
diff -urN util-vserver-0.30.216-r2834.orig/distrib/f10/pubkeys/RPM-GPG-KEY-fedora-test-10-primary util-vserver-0.30.216-r2834/distrib/f10/pubkeys/RPM-GPG-KEY-fedora-test-10-primary
--- util-vserver-0.30.216-r2834.orig/distrib/f10/pubkeys/RPM-GPG-KEY-fedora-test-10-primary	1970-01-01 01:00:00.000000000 +0100
+++ util-vserver-0.30.216-r2834/distrib/f10/pubkeys/RPM-GPG-KEY-fedora-test-10-primary	2009-01-22 00:52:16.000000000 +0100
@@ -0,0 +1,41 @@
+-----BEGIN PGP PUBLIC KEY BLOCK-----
+Version: GnuPG v1.4.9 (GNU/Linux)
+
+mQGiBEi13W8RBADWLvpDV7NziJYLMfXJzOLCEHRYaeDNowm4NV6399dkOd0Meuep
+AdiWjq4dymgml7N5guT7OjzuMcZXYqOQ8ge7FR/M3m1nKS9XSZPCwWadgcDejRzd
+ybTFnho9MiVH4TT4ozmI1pMQ6UcMb2TAiFF9C6cvHlboauRGLAsKOiuW+wCg1NGE
++KZr2ZAc3OKOZgJVq/0sQQ0D/3c041/hvaOb848PKAArlUhyTDPwFgxjaqzc70pL
+RVmYydYa8jaPvop1q/HCmVYCOJZEHGtaDbzhJl8L/Qy1yfQzWeuer0NB1xOUPtKj
+BvXP/KKpR7Kvz06jxWLc5hOg6D4Mf6Xy4AetMRnkeMQZ0qUANj+WQvqd/QyV7/1z
+lLgsA/wIGWy9dMJPXOVLuT6pMBAMkVvQDh5dAbYhCI70rNVGJplpksNXpBWBEkSQ
+z5DRR6d0ADBtJ1LaZWL0BC9dirJOJJyMxaM4G21Mgoc+XAehbKtVBZeF5pXwb2ch
+f4xqsuo0MRv8pvkbOm5WPGagUygfgUWmCOAreRTpTmrBVleT2LQuRmVkb3JhICgx
+MCB0ZXN0aW5nKSA8ZmVkb3JhQGZlZG9yYXByb2plY3Qub3JnPohgBBMRAgAgBQJI
+td1vAhsDBgsJCAcDAgQVAggDBBYCAwECHgECF4AACgkQkqECPQuGJ05MHACcCkTz
+QbIa1W74Dixo1WQSShB8P44AoL0wTcLLNjmgStugcUEnJ/dg8VAxuQQNBEi13W8Q
+EADZI/3ySMJtnSDFYsUis9dezKNoi4IbuAiY/CdaMUPRg1rNGpkn/tz4LE9o0bke
+ktXMCs5Xohv6e1fEOvm3bZNFZYSGea7cwp6sKD6jootHzLPAtJs9uyGiZbAXx29Q
+11HHr66Gna/VdyVGNuRuy5Ygpbn7XXGO6hJci/lbvUOxi1Fui4sjQdnpLuH5mPrl
+aXVMZSonn5lqLhAAQc9igPQVjFmbzhZA6x9z57ZvdjOWCkV1EdOIQUrhS3r5gnCv
+vUCjLBhhKmKtKNLAlaaRXSP+/3el6GZiXnOFOECy7u7CFaR+Qb9w7IF+tuE9lIVB
+r8UQNHpAzIzdxg376fXYwpkNbCOWMzqL9Y3HBXkdt58XQ+sR6yHzRlCc2mh9H8Pk
+SkYHcWvPL8njwQzrHWPp0PZEjdImSQL+hfY2IOq/vfd5toCypPFoLen79K4YaB2x
+F9W9vXs7W5kIviWTS/MOPMt16a3Sen31NPSNQN9q4zXOkAcus9jn0DBs6mozBE0H
+ojd+59p5Gh1zTYmPmgtURd4bWgHKOGHgnY9TuvAPPR4AXC5zcBZWhi7hbXeLyjyc
++8BdP/oX/FNVsFDZfBEvie+8TFsC5H6PEb00CXSCoKX+6FzyoLH/ASO+tXaMcm3U
+HabRf67bm60RReWCmHrMyKpLg3rP2K4kjMpDX5dyVRoumwADBQ//cUVFskjAVJ+p
+eAefeay9gxHg7Y4vGxylMc8EsIhw5R3+uf0rpAT0iPdtRLJwt1LUC/kRavGNpeGP
+4zeDXo1OItkxsAfMmnX6M5asm/OVHypl/Ubthp2/CuJk4hMTOG6a03SJsAAbrSDO
+on6uF+fpetIH5J0zhSkQXQClRtQEPd54tgmKxNsRTkzz/3TSnlCsMILamsjrMSE+
+WWxAWKclLe2wXAHCfRwuI5Zen8r5+t3JJaYVtcxrw3jeqGjM+obaoDTAhx1jWY0P
+I6ulowaFdNj3ZXOrFu5cVCmSDhidDV5yABpObPtw1s1GhJpMJfkja3hQH/v8MRyo
+hWZVhYvgj6HACNsMXvGqUh8IgpaK9nSeCBJw5w6xwD8D8F3LSCrDUbXIu7QPJ3Ty
+A9sZdjtSJeCyOFd4rP9yxcOAwTu/cerev/riGpEhAiUa0WQoR57N5x1aQTs/OoJx
+o6KolUrEkUaYAwYFJjn+Lr0mzQzPFFqAPwjr2c9G6hgLy4C10fWY5cpxHwc1ayo0
+sVeEFfmtcxrk074tZU4kRfgjz2j+e9dGCdi423DBSau7DEjuQe+N8JUbMSr2+nvm
+za/jGDwA4hD7/SBohBzVJq7MsCm0pXXvimaviYP72ghrXTmVd2obU29h5tTDXObW
+88SChY+u1OrgnABGsBl8Q7hqOk7R3fyISQQYEQIACQUCSLXdbwIbDAAKCRCSoQI9
+C4YnTkQyAJ9R/UthQ4rWFEZfNfZp7Glgo6e02QCfaGiQ0wvRd1xIME6O6HTsEOlC
+MC0=
+=we9R
+-----END PGP PUBLIC KEY BLOCK-----
diff -urN util-vserver-0.30.216-r2834.orig/distrib/f10/rpmlist.d/00.lst util-vserver-0.30.216-r2834/distrib/f10/rpmlist.d/00.lst
--- util-vserver-0.30.216-r2834.orig/distrib/f10/rpmlist.d/00.lst	1970-01-01 01:00:00.000000000 +0100
+++ util-vserver-0.30.216-r2834/distrib/f10/rpmlist.d/00.lst	2009-04-27 17:02:41.000000000 +0200
@@ -0,0 +1,25 @@
+basesystem-*.rpm
+filesystem-*.rpm
+glibc-common-*.rpm
+glibc-[0-9]*.rpm
+libgcc-[0-9]*.rpm
+setup-*.rpm
+tzdata-*.rpm
+bash-[0-9]*.rpm
+ncurses-libs-*.rpm
+ncurses-[0-9]*.rpm
+coreutils-[0-9]*.rpm
+ncurses-base-[0-9]*.rpm
+pam-[0-9]*.rpm
+libselinux-[0-9]*.rpm
+grep-[0-9]*.rpm
+info-[0-9]*.rpm
+libacl-[0-9]*.rpm
+cracklib-[0-9]*.rpm
+cracklib-dicts-[0-9]*.rpm
+audit-libs-[0-9]*.rpm
+libsepol-[0-9]*.rpm
+zlib-[0-9]*.rpm
+pcre-[0-9]*.rpm
+libattr-[0-9]*.rpm
+libstdc++-[0-9]*.rpm
diff -urN util-vserver-0.30.216-r2834.orig/distrib/f10/yum/yum.conf util-vserver-0.30.216-r2834/distrib/f10/yum/yum.conf
--- util-vserver-0.30.216-r2834.orig/distrib/f10/yum/yum.conf	1970-01-01 01:00:00.000000000 +0100
+++ util-vserver-0.30.216-r2834/distrib/f10/yum/yum.conf	2009-04-27 17:02:41.000000000 +0200
@@ -0,0 +1,11 @@
+[main]
+cachedir=@YUMCACHEDIR@
+persistdir=@YUMLIBDIR@
+reposdir=@YUMETCDIR@/yum.repos.d
+debuglevel=1
+logfile=@YUMLOGDIR@/log
+lockfile=@YUMLOCKDIR@/yum.pid
+pkgpolicy=newest
+distroverpkg=fedora-release
+installonlypkgs=
+exactarch=0
diff -urN util-vserver-0.30.216-r2834.orig/distrib/f10/yum.repos.d/fedora-development.repo util-vserver-0.30.216-r2834/distrib/f10/yum.repos.d/fedora-development.repo
--- util-vserver-0.30.216-r2834.orig/distrib/f10/yum.repos.d/fedora-development.repo	1970-01-01 01:00:00.000000000 +0100
+++ util-vserver-0.30.216-r2834/distrib/f10/yum.repos.d/fedora-development.repo	2009-04-27 17:16:30.000000000 +0200
@@ -0,0 +1,48 @@
+# These packages are untested and still under development. This
+# repository is used for updates to test releases, and for
+# development of new releases.
+#
+# This repository can see significant daily turnover and major
+# functionality changes which cause unexpected problems with other
+# development packages. Please use these packages if you want to work
+# with the Fedora developers by testing these new development packages.
+#
+# fedora-test-list@redhat.com is available as a discussion forum for
+# testing and troubleshooting for development packages in conjunction
+# with new test releases.
+#
+# More information is available at http://fedoraproject.org/wiki/Testing 
+#
+# Reproducible and reportable issues should be filed at
+# http://bugzilla.redhat.com/.
+#
+# Product: Fedora
+# Version: devel
+
+[development]
+name=Fedora - Development
+failovermethod=priority
+#baseurl=http://download.fedora.redhat.com/pub/fedora/linux/development/$basearch/os/
+mirrorlist=http://mirrors.fedoraproject.org/mirrorlist?repo=rawhide&arch=$basearch
+enabled=0
+gpgcheck=0
+gpgkey=http://redhat.download.fedoraproject.org/pub/fedora/linux/releases/10/Fedora/$basearch/os/RPM-GPG-KEY-fedora-test http://redhat.download.fedoraproject.org/pub/fedora/linux/releases/10/Fedora/$basearch/os/RPM-GPG-KEY-fedora
+
+[development-debuginfo]
+name=Fedora - Development - Debug
+failovermethod=priority
+#baseurl=http://download.fedora.redhat.com/pub/fedora/linux/development/$basearch/debug/
+mirrorlist=http://mirrors.fedoraproject.org/mirrorlist?repo=rawhide-debug&arch=$basearch
+enabled=0
+gpgcheck=0
+gpgkey=http://redhat.download.fedoraproject.org/pub/fedora/linux/releases/10/Fedora/$basearch/os/RPM-GPG-KEY-fedora-test http://redhat.download.fedoraproject.org/pub/fedora/linux/releases/10/Fedora/$basearch/os/RPM-GPG-KEY-fedora
+
+[development-source]
+name=Fedora - Development - Source
+failovermethod=priority
+#baseurl=http://download.fedora.redhat.com/pub/fedora/linux/development/source/SRPMS/
+mirrorlist=http://mirrors.fedoraproject.org/mirrorlist?repo=rawhide-source&arch=$basearch
+enabled=0
+gpgcheck=0
+gpgkey=http://redhat.download.fedoraproject.org/pub/fedora/linux/releases/10/Fedora/$basearch/os/RPM-GPG-KEY-fedora-test http://redhat.download.fedoraproject.org/pub/fedora/linux/releases/10/Fedora/$basearch/os/RPM-GPG-KEY-fedora
+
diff -urN util-vserver-0.30.216-r2834.orig/distrib/f10/yum.repos.d/fedora.repo util-vserver-0.30.216-r2834/distrib/f10/yum.repos.d/fedora.repo
--- util-vserver-0.30.216-r2834.orig/distrib/f10/yum.repos.d/fedora.repo	1970-01-01 01:00:00.000000000 +0100
+++ util-vserver-0.30.216-r2834/distrib/f10/yum.repos.d/fedora.repo	2009-04-27 17:18:32.000000000 +0200
@@ -0,0 +1,26 @@
+[fedora]
+name=Fedora 10 - $basearch
+failovermethod=priority
+#baseurl=http://download.fedora.redhat.com/pub/fedora/linux/releases/10/Everything/$basearch/os/
+mirrorlist=http://mirrors.fedoraproject.org/mirrorlist?repo=fedora-10&arch=$basearch
+enabled=1
+gpgcheck=1
+gpgkey=http://redhat.download.fedoraproject.org/pub/fedora/linux/releases/10/Fedora/$basearch/os/RPM-GPG-KEY-fedora http://redhat.download.fedoraproject.org/pub/fedora/linux/releases/10/Fedora/$basearch/os/RPM-GPG-KEY
+
+[fedora-debuginfo]
+name=Fedora 10 - $basearch - Debug
+failovermethod=priority
+#baseurl=http://download.fedora.redhat.com/pub/fedora/linux/releases/10/Everything/$basearch/debug/
+mirrorlist=http://mirrors.fedoraproject.org/mirrorlist?repo=fedora-debug-10&arch=$basearch
+enabled=0
+gpgcheck=1
+gpgkey=http://redhat.download.fedoraproject.org/pub/fedora/linux/releases/10/Fedora/$basearch/os/RPM-GPG-KEY-fedora http://redhat.download.fedoraproject.org/pub/fedora/linux/releases/10/Fedora/$basearch/os/RPM-GPG-KEY
+
+[fedora-source]
+name=Fedora 10 - Source
+failovermethod=priority
+#baseurl=http://download.fedora.redhat.com/pub/fedora/linux/releases/10/Everything/source/SRPMS/
+mirrorlist=http://mirrors.fedoraproject.org/mirrorlist?repo=fedora-source-10&arch=$basearch
+enabled=0
+gpgcheck=1
+gpgkey=http://redhat.download.fedoraproject.org/pub/fedora/linux/releases/10/Fedora/$basearch/os/RPM-GPG-KEY-fedora http://redhat.download.fedoraproject.org/pub/fedora/linux/releases/10/Fedora/$basearch/os/RPM-GPG-KEY
diff -urN util-vserver-0.30.216-r2834.orig/distrib/f10/yum.repos.d/fedora-updates.repo util-vserver-0.30.216-r2834/distrib/f10/yum.repos.d/fedora-updates.repo
--- util-vserver-0.30.216-r2834.orig/distrib/f10/yum.repos.d/fedora-updates.repo	1970-01-01 01:00:00.000000000 +0100
+++ util-vserver-0.30.216-r2834/distrib/f10/yum.repos.d/fedora-updates.repo	2009-04-27 17:28:53.000000000 +0200
@@ -0,0 +1,26 @@
+[updates]
+name=Fedora 10 - $basearch - Updates
+failovermethod=priority
+#baseurl=http://download.fedora.redhat.com/pub/fedora/linux/updates/10/$basearch/
+mirrorlist=http://mirrors.fedoraproject.org/mirrorlist?repo=updates-released-f10&arch=$basearch
+enabled=1
+gpgcheck=1
+gpgkey=http://redhat.download.fedoraproject.org/pub/fedora/linux/releases/10/Fedora/$basearch/os/RPM-GPG-KEY-fedora
+
+[updates-debuginfo]
+name=Fedora 10 - $basearch - Updates - Debug
+failovermethod=priority
+#baseurl=http://download.fedora.redhat.com/pub/fedora/linux/updates/10/$basearch/debug/
+mirrorlist=http://mirrors.fedoraproject.org/mirrorlist?repo=updates-released-debug-f10&arch=$basearch
+enabled=0
+gpgcheck=1
+gpgkey=http://redhat.download.fedoraproject.org/pub/fedora/linux/releases/10/Fedora/$basearch/os/RPM-GPG-KEY-fedora
+
+[updates-source]
+name=Fedora 10 - Updates Source
+failovermethod=priority
+#baseurl=http://download.fedora.redhat.com/pub/fedora/linux/updates/10/SRPMS/
+mirrorlist=http://mirrors.fedoraproject.org/mirrorlist?repo=updates-released-source-f10&arch=$basearch
+enabled=0
+gpgcheck=1
+gpgkey=http://redhat.download.fedoraproject.org/pub/fedora/linux/releases/10/Fedora/$basearch/os/RPM-GPG-KEY-fedora
diff -urN util-vserver-0.30.216-r2834.orig/distrib/f10/yum.repos.d/fedora-updates-testing.repo util-vserver-0.30.216-r2834/distrib/f10/yum.repos.d/fedora-updates-testing.repo
--- util-vserver-0.30.216-r2834.orig/distrib/f10/yum.repos.d/fedora-updates-testing.repo	1970-01-01 01:00:00.000000000 +0100
+++ util-vserver-0.30.216-r2834/distrib/f10/yum.repos.d/fedora-updates-testing.repo	2009-04-27 17:28:47.000000000 +0200
@@ -0,0 +1,26 @@
+[updates-testing]
+name=Fedora 10 - $basearch - Test Updates
+failovermethod=priority
+#baseurl=http://download.fedora.redhat.com/pub/fedora/linux/updates/testing/10/$basearch/
+mirrorlist=http://mirrors.fedoraproject.org/mirrorlist?repo=updates-testing-f10&arch=$basearch
+enabled=0
+gpgcheck=1
+gpgkey=http://redhat.download.fedoraproject.org/pub/fedora/linux/releases/10/Fedora/$basearch/os/RPM-GPG-KEY-fedora-test
+
+[updates-testing-debuginfo]
+name=Fedora 10 - $basearch - Test Updates Debug
+failovermethod=priority
+#baseurl=http://download.fedora.redhat.com/pub/fedora/linux/updates/testing/10/$basearch/debug/
+mirrorlist=http://mirrors.fedoraproject.org/mirrorlist?repo=updates-testing-debug-f10&arch=$basearch
+enabled=0
+gpgcheck=1
+gpgkey=http://redhat.download.fedoraproject.org/pub/fedora/linux/releases/10/Fedora/$basearch/os/RPM-GPG-KEY-fedora-test
+
+[updates-testing-source]
+name=Fedora 10 - Test Updates Source
+failovermethod=priority
+#baseurl=http://download.fedora.redhat.com/pub/fedora/linux/core/updates/testing/10/SRPMS/
+mirrorlist=http://mirrors.fedoraproject.org/mirrorlist?repo=updates-testing-source-f10&arch=$basearch
+enabled=0
+gpgcheck=1
+gpgkey=http://redhat.download.fedoraproject.org/pub/fedora/linux/releases/10/Fedora/$basearch/os/RPM-GPG-KEY-fedora-test
diff -urN util-vserver-0.30.216-r2834.orig/distrib/Makefile.am util-vserver-0.30.216-r2834/distrib/Makefile.am
--- util-vserver-0.30.216-r2834.orig/distrib/Makefile.am	2009-04-27 17:40:58.000000000 +0200
+++ util-vserver-0.30.216-r2834/distrib/Makefile.am	2009-04-27 17:43:21.000000000 +0200
@@ -73,6 +73,10 @@
 			   $(wildcard $(srcdir)/f9/yum.repos.d/*.repo) \
 			   $(wildcard $(srcdir)/f9/rpmlist.d/*.lst) $(wildcard $(srcdir)/9/rpmlist.d/*.opt) \
 			   $(wildcard $(srcdir)/f9/pubkeys/RPM-GPG*) \
+			   f10/pkgs/01 f10/pkgs/02 f10/pkgs/03 f10/yum/yum.conf \
+			   $(wildcard $(srcdir)/f10/yum.repos.d/*.repo) \
+			   $(wildcard $(srcdir)/f10/rpmlist.d/*.lst) $(wildcard $(srcdir)/9/rpmlist.d/*.opt) \
+			   $(wildcard $(srcdir)/f10/pubkeys/RPM-GPG*) \
 			   centos4/pkgs/01 centos4/pkgs/02 centos4/pkgs/03 \
 			   centos4/yum/yum.conf $(wildcard $(srcdir)/centos4/yum.repos.d/*.repo) \
 			   centos5/pkgs/01 centos5/pkgs/02 centos5/pkgs/03 \
@@ -81,11 +85,6 @@
 			   suse91/rpm/macros \
 			   gentoo/init-vserver.sh gentoo/net.vserver \
 			   gentoo/reboot.sh gentoo/shutdown.sh \
-			   debian/debootstrap.script \
-			   breezy/debootstrap.script dapper/debootstrap.script \
-			   edgy/debootstrap.script   feisty/debootstrap.script \
-			   gutsy/debootstrap.script  hoary/debootstrap.script \
-			   warty/debootstrap.script \
 			   pld-ac/pkgs/01 pld-ac/rpm/macros \
 			   pld-th/pkgs/01 pld-th/rpm/macros \
 			   pld-ti/pkgs/01 pld-ti/rpm/macros
@@ -111,7 +110,7 @@
 			   $(nobase_confdistrib_DATA) \
 			   $(defaults_DATA)
 
-redhat_style		=  rh9 fc1 fc2 fc3 fc4 fc5 fc6 f7 f8 f9 centos4 centos5
+redhat_style		=  rh9 fc1 fc2 fc3 fc4 fc5 fc6 f7 f8 f9 f10 centos4 centos5
 sid_style		=  etch lenny sid
 
 install-exec-hook:	install-notify-xtra
