--- util-vserver-0.30.212/scripts/vrpm-preload~	2007-11-07 12:24:38.000000000 +0200
+++ util-vserver-0.30.212/scripts/vrpm-preload	2007-11-07 12:25:09.260798172 +0200
@@ -27,7 +27,7 @@
 . $__PKGLIBDIR/vserver.functions
 . "$_LIB_FUNCTIONS"
 
-for bin in `which rpm` /bin/rpm /usr/lib/rpm/rpm${RPM_FLAVOR:-i} ""; do
+for bin in /bin/rpm /usr/lib/rpm/rpm${RPM_FLAVOR:-i} ""; do
     ldd "$bin" &>/dev/null && break
 done
 
