--- util-vserver-0.30.214/scripts/vserver.functions~	2007-09-02 23:06:48.000000000 +0200
+++ util-vserver-0.30.214/scripts/vserver.functions	2007-09-02 23:14:40.713357250 +0200
@@ -284,7 +284,7 @@
 		INITCMD_START=( "$i" "$RUNLEVEL_START" )
 		INITCMD_STOP=(  "$i" "$RUNLEVEL_STOP"  )
 	    done
-	    INITCMD_PREPARE=( $_FAKE_RUNLEVEL "$RUNLEVEL_START" /var/run/utmp )
+	    INITCMD_PREPARE=( $_FAKE_RUNLEVEL "$RUNLEVEL_START" /var/run/utmpx )
 	    OPTS_ENV=( "${OPTS_ENV[@]}" PREVLEVEL=N RUNLEVEL="$RUNLEVEL_START" )
 	    if test -n "$OPTION_DEBUG_SYSV"; then
 		INITCMD_START=( /bin/bash -x "${INITCMD_START[@]}" )
@@ -309,7 +309,7 @@
 	    _NEED_VSHELPER_SYNC=1
 	    test -z "$RUNLEVEL_START"         || INITCMD_START=( "${INITCMD_START[@]}" "$RUNLEVEL_START" )
 	    test -z "$RUNLEVEL_STOP"          || INITCMD_STOP=(  "${INITCMD_STOP[@]}"  "$RUNLEVEL_STOP"  )
-	    ! isNumber "${RUNLEVEL_START:-3}" || INITCMD_PREPARE=( $_FAKE_RUNLEVEL "${RUNLEVEL_START:-3}" /var/run/utmp )
+	    ! isNumber "${RUNLEVEL_START:-3}" || INITCMD_PREPARE=( $_FAKE_RUNLEVEL "${RUNLEVEL_START:-3}" /var/run/utmpx )
 	    ;;
 
 	(xgentoo)
@@ -317,7 +317,7 @@
 
 	    INITCMD_START=( env TERM=$TERM $RC_WRAP "$RUNLEVEL_START" )
 	    INITCMD_STOP=( env -i PATH=$RC_PATH TERM=$TERM RUNLEVEL=0 /sbin/rc shutdown )
-	    INITCMD_PREPARE=( $_FAKE_RUNLEVEL 3 /var/run/utmp )
+	    INITCMD_PREPARE=( $_FAKE_RUNLEVEL 3 /var/run/utmpx )
 	    ;;
 
 	(x) ;;
