--- configure.ac~	2009-06-26 02:31:27.000000000 +0200
+++ configure.ac	2009-07-28 12:57:13.562148646 +0200
@@ -250,6 +250,7 @@
 IFS=,;
 
 for i in $enable_apis; do
+	IFS=$old_IFS
 	case "$i" in
 		(compat)	AC_DEFINE(VC_ENABLE_API_COMPAT,   1)
 				enable_api_oldproc=${enable_api_oldproc:-1}
@@ -276,6 +277,7 @@
 		(olduts)	enable_api_olduts=2;;
 		(*)		AC_MSG_ERROR(['$i' is not a supported API]);;
 	esac
+	IFS=","
 done
 IFS=$old_IFS
 if test x"$enable_api_oldproc" != x; then
