diff -urN util-vserver-0.30.210.org/kernel/network.h util-vserver-0.30.210/kernel/network.h
--- util-vserver-0.30.210.org/kernel/network.h	2006-01-02 23:16:34.000000000 +0100
+++ util-vserver-0.30.210/kernel/network.h	2006-05-15 10:18:29.390612500 +0200
@@ -8,7 +8,7 @@
 
 #define NX_DYNAMIC_ID	((uint32_t)-1)		/* id for dynamic context */
 
-#define NB_IPV4ROOT	16
+#define NB_IPV4ROOT	128
 
 
 /* network flags */
diff -urN util-vserver-0.30.210.org/src/chbind.c util-vserver-0.30.210/src/chbind.c
--- util-vserver-0.30.210.org/src/chbind.c	2006-01-22 19:17:24.000000000 +0100
+++ util-vserver-0.30.210/src/chbind.c	2006-05-15 10:18:00.285703875 +0200
@@ -322,7 +322,7 @@
 #endif
       case CMD_IP		:
 	if (nbaddrs>=nb_ipv4root) {
-	  WRITE_MSG(2, "Too many IP numbers, max 16\n");
+	  WRITE_MSG(2, "Too many IP numbers, max 128\n");
 	  exit(wrapper_exit_code);
 	}
 	readIP(optarg, ips+nbaddrs, &bcast);
