--- util-vserver-0.30.215/scripts/vserver-build.functions~	2008-11-21 18:36:30.000000000 +0100
+++ util-vserver-0.30.215/scripts/vserver-build.functions	2008-11-21 18:47:00.000000000 +0100
@@ -57,6 +57,10 @@
     mkdir -p -m755 "$VDIR"/dev
     mkdir -m755 "$VDIR"/dev/pts
 
+    ln -s /proc/self/fd/0 "$VDIR"/dev/stdin
+    ln -s /proc/self/fd/1 "$VDIR"/dev/stdout
+    ln -s /proc/self/fd/2 "$VDIR"/dev/stderr
+
     while read spec; do
 	makeDevEntry "$VDIR"/dev $spec
     done <$_DEV_FILE
