diff -ur util-vserver-0.30.210.orig/scripts/vserver-build util-vserver-0.30.210/scripts/vserver-build
--- util-vserver-0.30.210.orig/scripts/vserver-build	2006-04-18 18:10:30.000000000 +0200
+++ util-vserver-0.30.210/scripts/vserver-build	2006-04-18 18:37:37.000000000 +0200
@@ -122,6 +122,9 @@
     shift
 done
 
+test -n "$SETUP_CONTEXT" ||
+    panic $"Dynamic Context IDs not supported (use --context option)"
+
 test -n "$VSERVER_NAME" ||
     panic $"Name of vserver not specified"
 
diff -ur util-vserver-0.30.210.orig/scripts/vserver.functions util-vserver-0.30.210/scripts/vserver.functions
--- util-vserver-0.30.210.orig/scripts/vserver.functions	2006-04-18 18:10:30.000000000 +0200
+++ util-vserver-0.30.210/scripts/vserver.functions	2006-04-18 19:04:07.000000000 +0200
@@ -365,6 +365,10 @@
 
     test -z "$S_CONTEXT" || ctx=$S_CONTEXT
 
+    test -n "$ctx" || panic $"\
+Dynamic Context IDs are not supported, you must set Context ID
+in $vdir/context file"
+
     _generateCapabilityOptions "$vdir"
     _generateFlagOptions       "$vdir"
 
