diff -urNp -x '*.orig' util-vserver-0.30.216-pre3126.org/scripts/vserver-build util-vserver-0.30.216-pre3126/scripts/vserver-build
--- util-vserver-0.30.216-pre3126.org/scripts/vserver-build	2021-03-12 21:50:27.621046552 +0100
+++ util-vserver-0.30.216-pre3126/scripts/vserver-build	2021-03-12 21:50:27.707714756 +0100
@@ -138,6 +138,9 @@ while true; do
     shift
 done
 
+test -n "$SETUP_CONTEXT" ||
+    panic $"Dynamic Context IDs not supported (use --context option)"
+
 test -n "$VSERVER_NAME" ||
     panic $"Name of vserver not specified"
 
diff -urNp -x '*.orig' util-vserver-0.30.216-pre3126.org/scripts/vserver.functions util-vserver-0.30.216-pre3126/scripts/vserver.functions
--- util-vserver-0.30.216-pre3126.org/scripts/vserver.functions	2016-02-24 19:09:17.000000000 +0100
+++ util-vserver-0.30.216-pre3126/scripts/vserver.functions	2021-03-12 21:50:27.707714756 +0100
@@ -455,6 +455,10 @@ function _generateChcontextOptions
 
     test -z "$S_CONTEXT" || ctx=$S_CONTEXT
 
+    test -n "$ctx" || panic $"\
+Dynamic Context IDs are not supported, you must set Context ID
+in $vdir/context file"
+
     _generateCapabilityOptions "$vdir"
     _generateFlagOptions       "$vdir"
     _generateUMaskOptions      "$vdir"
